lib/internal/fs/streams.js

```diff
@@ -339,7 +354,14 @@ WriteStream.prototype._write = function(data, encoding, cb) {
     });
   }
 
+  if (this.destroyed) return;
+
+  this.isPerformingIO = true;
   fs.write(this.fd, data, 0, data.length, this.pos, (er, bytes) => {
+    this.isPerformingIO = false;
+    // Tell ._destroy() that it's safe to close the fd now.
+    if (this.destroyed) return this.emit(kIoDone);
```

Comment:
This will swallow the error and deadlock a `write(cb)` call (i.e. never invoke the callback).

Reply:
```js
if (this.destroyed) {
  cb(er); // Don't swallow error and ensure callback is invoked.
  this.emit(kIoDone);
  return
}
```